bindings/gumjs/gumscript-runtime-dalvik.js

```diff
@@ -721,14 +721,19 @@
                 });
 
                 var implementation = null;
-                var synchronizeVtable = function (env) {
+                var synchronizeVtable = function (env, instance) {
                     if (originalMethodId === null) {
                         return; // nothing to do – implementation hasn't been replaced
                     }
 
                     var thread = Memory.readPointer(env.handle.add(JNI_ENV_OFFSET_SELF));
-                    var object = api.dvmDecodeIndirectRef(thread, this.$handle);
-                    var classObject = Memory.readPointer(object.add(OBJECT_OFFSET_CLAZZ));
+                    var objectPtr = api.dvmDecodeIndirectRef(thread, instance? this.$handle: this.$classHandle);
```

Comment:
For consistency it would be good to use a space also on the left side of the `?` and `:`.